packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -219,12 +219,21 @@ class UpgradeCommandRunner {
           workingDirectory: workingDirectory,
       );
       revision = result.stdout.trim();
-    } on Exception {
-      throwToolExit(
-        'Unable to upgrade Flutter: no origin repository configured. '
-        "Run 'git remote add origin "
-        "https://github.com/flutter/flutter' in $workingDirectory",
-      );
+    } on Exception catch (e){
+      final String errorString = e.toString();
+      if (RegExp(r'fatal: HEAD does not point to a branch').hasMatch(errorString)) {
+        throwToolExit(
+          'You are not currently on a release branch. Use git to '
+          'check out an official branch\n(\'stable\', \'beta\', \'dev\', or \'master\') '
+          'and retry, for example \'git checkout stable\'.'
```

Comment:
Based on an audit of user messages, how about:
```
'You are not currently on a release branch. Use git to '
'check out an official branch (\'stable\', \'beta\', \'dev\', or \'master\') '
'and retry, for example:\n'
'  git checkout stable'
```

Reply:
done